runtime/parachains/src/inclusion_inherent.rs

```diff
@@ -107,7 +107,9 @@ decl_module! {
 
 			// Process new availability bitfields, yielding any availability cores whose
 			// work has now concluded.
+			let expected_bits = <scheduler::Module<T>>::availability_cores().len();
 			let freed_concluded = <inclusion::Module<T>>::process_bitfields(
+				expected_bits,
```

Comment:
this was previously
```
Paras::parachains().len() + Configuration::config().parathread_cores as usize
```
could you elaborate why?

Reply:
Yes. It's because of `max_validators_per_core` in the `Scheduler` module. (#2043 and #2143). There can actually be up to `n_validators / max_validators_per_core` cores, although any beyond `n_parachains + n_parathreads` will be perpetually Free.

This calculation was wrong and it's better to pull the value directly from the scheduler as we should compute this value in just one place.